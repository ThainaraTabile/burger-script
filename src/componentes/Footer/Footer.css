
footer{
    width: 100%;
    margin-top: 4rem;
    max-height: 10rem;
}

.img-footer{
    max-height: 20rem;
}