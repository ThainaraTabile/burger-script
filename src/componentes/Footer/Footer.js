import React from 'react';
import './Footer.css';

const Footer = ({ imagemSrc, imagemAlt }) => {
  return (
    <footer data-testid='footer'>
      <img src={imagemSrc} alt={imagemAlt} className="img-footer" />
    </footer>
  );
};

export default Footer;
