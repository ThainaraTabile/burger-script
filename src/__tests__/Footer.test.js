import Footer from "../componentes/Footer/Footer";
import { expect } from '@jest/globals';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

describe('Footer', () => {

    test('Verifica se o componente é renderizado corretamente', () => {
        render(<Footer />);
        const componenteFooter = screen.getByTestId('footer');
        expect(componenteFooter).toBeInTheDocument();
        expect(componenteFooter).toMatchSnapshot();

    });

    test('Verifica se a imagem, atributos src e alt estão sendo renderizados corretamente', () => {
        const imagemSrc = 'icone-teste.webp';
        const imagemAlt = 'Ícone de teste';

        render(<Footer imagemSrc={imagemSrc} imagemAlt={imagemAlt} />);

        const imagemElement = screen.getByAltText(imagemAlt);
        expect(imagemElement).toBeInTheDocument();
        expect(imagemElement).toHaveAttribute('src', expect.stringContaining(imagemSrc));
    });

})