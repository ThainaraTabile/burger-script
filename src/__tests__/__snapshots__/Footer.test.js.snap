// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Verifica se o componente Ã© renderizado corretamente 1`] = `
<footer
  data-testid="footer"
>
  <img
    class="img-footer"
  />
</footer>
`;
